modules/average_month.py

```diff
@@ -0,0 +1,109 @@
+"""
+This module generates a report which displays average highest and lowest temperature
+as well as average max humidity for a given month
+"""
+import constants
+from modules.utils import (
+    get_highest_temperature,
+    get_lowest_temperature,
+    get_mean_humidity,
+    read_data,
+)
+
+
+def calculate_averages(pattern, path):
+    """
+    This function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. avg highest temperature
+    2. avg lowest temperature
+    3. avg mean humidity
+    Args:
+        pattern(str): a string which contains a year/month e.g '2006/5', '2007/3'
+        path(str): a string which contains path to a directory containing weather files
+    Returns:
+        (list or None): list of length 3, averages where
+                        averages[0] = avg_highest_temperature
+                        averages[1] = avg_lowest_temperature
+                        averages[2] = avg_mean_humidity
+                        Or
+                        None for failure
+    """
+    sum_highest_temperature = 0
+    sum_lowest_temperature = 0
+    sum_mean_humidity = 0
+    lines = []
+    for lines in read_data(pattern, path):
+        if lines is None:
+            return None
+        for line in lines:
+            parsed_line = line.split("\n")[0].split(",")
+            max_temperature = get_highest_temperature(parsed_line)
+            if max_temperature is not None:
+                sum_highest_temperature += max_temperature
+            min_temperature = get_lowest_temperature(parsed_line)
+            if min_temperature is not None:
+                sum_lowest_temperature += min_temperature
+            mean_humidity = get_mean_humidity(parsed_line)
+            if mean_humidity is not None:
+                sum_mean_humidity += mean_humidity
```

Comment:
Can you try this? You might not need to use a None check above it. This would apply to all the other places as well.

```suggestion
                sum_mean_humidity += mean_humidity or 0
```

Reply:
this is great. applied to all 3 sums